### ISSUE TITLE
Aseprite importer: Add importer to rooms and props

### ISSUE THREAD

**Issue by stickgrinder** [mapedorr/popochiu#37]
Related to #8 

Importer must work for Rooms and Props also.
This issue has to be detailed.

Evaluate a function to reload the single Animation (this is in PQ but may make little sense for we are stuck with animations atlases - Maybe good for props...)

**Comment by stickgrinder** [comment]
Working on this issue, as is visible from the long list of commits, forced me to move the Popochiu Props creation logic out of the creation popup. The reason is that before this replacement, the only way to create a prop was interactively, via the popup and human interaction. In fact, I need a programmatic way to create props in rooms for each tag in an Aseprite source file.

To avoid leaving half the code like this and half the code like that, I embarked on moving everything. I did and it works but during the process, I decided some more love would have made things better, so I am refactoring the old code as follows:

### Things to refactor to call this code clean

* [ ] Refactor all creation logic in a class with subclasses
  * [ ] Character and Rooms at the moment only differ from a template and a placeholder in the base script template (see `popochiu_character_helper.gd` and `popochiu_room_helper.gd` lines `44:72` for example). Since that's true for the other objects too, I would refactor this all to the base class.
  * Same is for the bulk of other functions. Let's make them dependent on a set of private class properties that the each specialized class must populate in its constructor. This will probably reduce code by 80%.
  * [ ] In script templates for various popochiu objs we have placeholders like `character_state_template`, `character_%s_state`, `room_state_template`, `room_%s_state`, `item_state_template`, ... and so on. We can clean the code up entirely if we just switch to `obj_state_template` and `ojb_%s_state`. (Bonus: we can refactor the templates in a main header + partials for the specific objects)
  * [ ] In main objects popup the `_obj_path_template` has to be declared somehow in the helper or whatever... maybe it is already, in the constructor/init? Have to check
  * [ ] Change helpers init function to a real constructor, why two calls? (because I'm dumb)
  * [ ] Instead of returning the scene object just to open the script or scene in the editor, the best option is to give helpers public methods like `get_scene_path()`, `get_resource_path()` or `get_script_path()`. They can come in handy in many other occasions too!
  * [ ] Change the helpers classes to Factory classes (that's what they are, not helpers) and create a new folder for them.
  * [ ] Consistent naming everywhere!

#### Suggestions by @mapedorr 

  * [ ] We can get rid of `_room_tab = _main_dock.get_opened_room_tab()` in `popochiu_room_obj_base_helper.gd` and replace it by a function. That will allow us to remove the `_room_tab` variable. At the moment we're using it only to add the created room object to the proper list in the Room tab. **SEEMS LEGIT, also relates to #67**
  * [ ] Maybe we can move the `const BASE_???` vars to `popochiu_resources.gd` (or take those paths from there if they already exisits) to have a better understading of what they are pointing to when reading the code. E.g.
    > **popochiu_character_helper.gd**
    > ```
    > # Instead of having
    > load(BASE_STATE_TEMPLATE).duplicate()
    > 
    > # We could have
    > load(popochiu_resources.CHARACTER_STATE_TEMPLATE).duplicate()
    > ```
  **THERE ARE CAVEATS I will explain in a later comment**

### The following are candidates for new issues

* [ ] Remove the need to pass the tab_room as soon as every popup doesn't need it (just the _main_dock) **Can't remember what I meant with this note, but removing the need to pass this variable is basically #67 so maybe this note is redundant**
* [ ] Clean up the old pascal_case text updater in the base creation_popup class **Godot 4+ has methods for that**
* [ ] Use Popochiu types helpers for tests on types
* [ ] Provide more `is_TYPE` helpers?

These notes are here for future reference since the PR for this issue will be HUGE.

Yes, the scope creep skyrocketed here but we're applying the boy-scout rule: _leave the place better than you find it!_ :)

**Comment by stickgrinder** [comment]
@mapedorr, reviewing your suggestions in light of the work I did, I'm leaving here my considerations:

* I heavily reduced the surface where the `_room_tab` update is carried out. This should make things more maintainable in the meantime. Maybe I don't understand what you mean by creating a function (where?)  but to me the best option is to leave it as it is for now and take #67 over to clean each and every place in the editor plugin where we need to pass docks and open tabs over. This is probably the cleanest option to make it all very loosely coupled :)
* About moving constants to `PopochiuResources`, it makes sense but before getting to it, please, take a look at what I achieved in terms of code deduplication: using a constant template and configuring each factory in its constructor (they are not stateful classes and private methods mostly act over the internal state, as you can see), there is no (less?) need of having many different "per type" constants. One thing we can do is using constants in `PopochiuResources` to populate state variables in the constructors, so this

```gdscript
	_obj_type = Constants.Types.PROP
	_obj_type_label = 'prop'
	_obj_room_group = 'Props'
	_obj_path_template = '/props/%s/prop_%s'
```

becomes this

```gdscript
	_obj_type = Constants.Types.PROP
	_obj_type_label = Constants.TypeLabels.PROP
	_obj_room_group = Constants.RoomObjectsGroups.PROP
	_obj_path_template = Constants.TypePathsTemplates.PROP
```

Now, this sucks because I made it up on the spot, but this is probably the outcome of moving such stuff.

Now that things are way more coherent between different objects, I will try to reduce the need for so many variables and maybe (as a middle ground), use local constants to populate them, just to make it more robust.

Let me know if you think those two points are still relevant, given the current implementation. Thanks.

**Comment by stickgrinder** [comment]
This one has been merged to 2.0 branch.
Closing this. Fixes and polishing will be tracked if necessary, into new features.
